include/ginkgo/core/base/array.hpp

```diff
@@ -637,6 +637,46 @@ class array {
      */
     const value_type* get_const_data() const noexcept { return data_.get(); }
 
+    /**
+     * Returns a single value from the array.
+     *
+     * This involves a bounds check, polymorphic calls and potentially a
+     * device-to-host copy, so it is not suitable for accessing many elements
+     * in performance-critical code.
+     *
+     * @param index  the array element index.
+     * @return the value at index.
+     */
+    value_type get_value(size_type index) const
```

Comment:
Maybe change the name to make sure it is clear that it can be copied from a device to the host.
```suggestion
    value_type get_value_to_host(size_type index) const
```